lib/components/Geojson.js

```diff
@@ -159,6 +159,7 @@ const Geojson = props => {
               image={props.image}
               pinColor={markerColor}
               zIndex={props.zIndex}
+              onPress={() => props.onPress(overlay)}
```

Comment:
Do we need to return overlay here or can follow what we do [here](https://github.com/react-native-maps/react-native-maps/blob/e38f5e1e61058711d571fe0029a7dd9c2cf070b2/lib/components/Geojson.js#L176) ?

Reply:
Actually, I think we need to return overlay otherwise I cannot find a way to reach which marker did I press. If you suggest a better way of doing that, I can convert it.

Reply:
Are you using coordinates to determine what marker was pressed? The `onPress` on MapMarker should already be passing the coordinates from the native event back. https://github.com/react-native-maps/react-native-maps/blob/master/docs/marker.md#events  so you could just go with `onPress={props.onPress}`

Reply:
I tried that event but I could not get the overlay's data with that event. Simply it does not contain its data. That's why we need to return the pressed `overlay`. 

So, I did not use coordinates to determine what marker is pressed actually we should not do that, it is very redundant because we already have which marker and its data is pressed inside the GeoJSON's logic.

Reply:
![CleanShot 2021-03-02 at 9 38 34](https://user-images.githubusercontent.com/6813862/109697856-ccbeb980-7b9f-11eb-9358-e25475cb6444.png)

@christopherdro,
Here is the onPress event object, it simply does not contain anything.

Reply:
@WrathChaos Thanks for the explanation. I have very limited experience using the Geojson component. @fattomhk Any thoughts here in relation to the changes to the Polygon component? Would it be beneficial to expose overlay there as well?

Reply:
@christopherdro 
I agree that we need something to identify the clicked polygon like what we already have in maker's `onPress`

BTW,  I found that it is hard to customize the overlays with `Geojson`. In my project, I loop the geojson file and use the `Geojson` component for generating polygons.

Reply:
@fattomhk It is really hard to customize and use Geojson with interaction. We cannot even change the size of the marker with the geojson and these are really useful to customise it.

Reply:
Also, we need to add the press functionality for Polyline as well. I've made it on this project's other branch but waiting for this to resolve.

Reply:
@WrathChaos Thoughts on including those changes in this PR?

Reply:
@christopherdro Sure.

Reply:
@christopherdro @fattomhk Both `Polyline` and `Polygon` has the pressable feature now. Can you guys check it again? 

Btw, we're using this code in production right now. It is working well.